Functional.Primitives.Extensions/OptionExtensions.cs

```diff
@@ -59,6 +60,10 @@ public static Option<TValue> Where<TValue>(this Option<TValue> option, Func<TVal
 		public static async Task<Option<TValue>> Where<TValue>(this Task<Option<TValue>> option, Func<TValue, bool> predicate)
 			=> (await option).Where(predicate);
 
+		public static IEnumerable<TValue> ToEnumerable<TValue>(this Option<TValue> option) => option.Match(value => Enumerable.Repeat(value, 1), Enumerable.Empty<TValue>);
```

Comment:
What's the intended use case for this?

Reply:
Here's the relevant code I was able to find...
```
// x.GroupKey is Option<string>
// groupKey is Option<string>
sut.CancellationTokenCountByGroupKey.Should().Be(
	originalCollectionOfCacheItems.GroupBy(x => x.GroupKey.ValueOrDefault()).Select(x => x.Key).Union(groupKey.ToEnumerable()).Distinct().Count(),
	groupKey.Match(gk => originalCollectionOfCacheItems.Any(x => x.GroupKey.ValueOrDefault() == gk) ? "the group key is already in the cache" : "the group key was not already in the cache", () => "no group key was given"));
```

It's mostly for `someCollectionOfT.Union(optionOfT.ToEnumerable())` considering `WhereSome` extension already exists with a different implementation...  Otherwise, `WhereSome` could be the following:
```
public static IEnumerable<T> WhereSome<T>(this IEnumerable<Option<T>> source) => source.SelectMany(x => x.ToEnumerable());
```